lib/elixir/lib/exception.ex

```diff
@@ -1330,6 +1330,10 @@ defmodule Enum.EmptyError do
   defexception message: "empty error"
 end
 
+defmodule Enum.NotFoundError do
+  defexception message: "not found error"
+end
```

Comment:
Maybe we would benefit from a more clear message, including which value it failed to find on which input?

Reply:
Apart from that, I think a more clear message could be used

Reply:
@v0idpwn The issue is that we are searching based on an anonymous function, not a value, so I don't think it would be simple to include "what we are searching for" in the string message. Additionally, the enumerable could be huge, again making it impossible to include in the message.

Reply:
You're totally correct.